dns_reverse_proxy.go

```diff
@@ -124,8 +124,10 @@ func route(w dns.ResponseWriter, req *dns.Msg) {
 		dns.HandleFailed(w, req)
 		return
 	}
+
+	lcName := strings.ToLower(req.Question[0].Name)
 	for name, addrs := range routes {
-		if strings.HasSuffix(req.Question[0].Name, name) {
+		if strings.HasSuffix(lcName, name) {
```

Comment:
routes could have been configured in uppercase, can you add:
`name := strings.ToLower(name)`